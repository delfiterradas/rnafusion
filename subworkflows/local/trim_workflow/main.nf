

include { FASTP                      } from '../../../modules/nf-core/fastp/main'
include { FASTQC as FASTQC_FOR_FASTP } from '../../../modules/nf-core/fastqc/main'

workflow TRIM_WORKFLOW {

    take:
        reads           // channel [ meta, [ fastq files ] ]
        adapter_fasta   // channel [ path ]
        fastp_trim      // boolean

    main:
        ch_versions       = Channel.empty()
        ch_fastp_html     = Channel.empty()
        ch_fastp_json     = Channel.empty()
        ch_fastqc_trimmed = Channel.empty()

        if ( fastp_trim ) {
            FASTP(reads, adapter_fasta.ifEmpty( [] ), false, false, false)
            ch_versions = ch_versions.mix(FASTP.out.versions)

            FASTQC_FOR_FASTP(FASTP.out.reads)
            ch_versions = ch_versions.mix(FASTQC_FOR_FASTP.out.versions)

            ch_reads_all           = FASTP.out.reads
            ch_reads_fusioncatcher = ch_reads_all
            ch_fastp_html          = FASTP.out.html
            ch_fastp_json          = FASTP.out.json
            ch_fastqc_trimmed      = FASTQC_FOR_FASTP.out.zip

        }
        else {
            ch_reads_all           = reads
            ch_reads_fusioncatcher = reads
        }

    emit:
        ch_reads_all            // Channel [ meta, [reads]   ]
        ch_reads_fusioncatcher  // Channel [ meta, [reads]   ]
        ch_fastp_html           // Channel [ meta, path_html ]
        ch_fastp_json           // Channel [ meta, path_json ]
        ch_fastqc_trimmed       // Channel [ meta, path_zip  ]
        versions = ch_versions  // Channel [ versions        ]
    }

