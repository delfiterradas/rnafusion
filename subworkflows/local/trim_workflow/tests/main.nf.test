nextflow_workflow {

    name "Test Subworkflow QC_WORKFLOW"
    script "../main.nf"
    workflow "TRIM_WORKFLOW"
    tag "qc"
    tag "subworkflow"
    tag "fastqc"
    tag "fastp"

    test("TRIM_WORKFLOW - Homo sapiens - FASTQs - fastp == true") {

        when {

            workflow {
                """
                // ch_reads
                input[0] = Channel.of(
                    [
                        [id: "test_fastq"],
                        [
                            file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/illumina/fastq/test_1.fastq.gz", checkIfExists: true),
                            file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/illumina/fastq/test_2.fastq.gz", checkIfExists: true)
                        ]
                    ])
                input[1] = Channel.empty()
                input[2] = true
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { with(workflow.out) {
                    assert snapshot(
                        versions.collect{ file(it) },
                        ch_reads_all[0][1].collect { file(it) },
                        ch_reads_fusioncatcher[0][1].collect { file(it) },
                        file(ch_fastp_html[0][1]).name,
                        file(ch_fastp_json[0][1]),
                        ch_fastqc_trimmed[0][1].collect { file(it).name }
                    ).match() }
                }
            )
        }
    }

    test("TRIM_WORKFLOW - Homo sapiens - FASTQs - fastp == true + fasp_adaptors") {

        when {

            workflow {
                """
                // ch_reads
                input[0] = Channel.of(
                    [
                        [id: "test_fastq"],
                        [
                            file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/illumina/fastq/test_1.fastq.gz", checkIfExists: true),
                            file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/illumina/fastq/test_2.fastq.gz", checkIfExists: true)
                        ]
                    ])
                input[1] = Channel.fromPath('https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/delete_me/fastp/adapters.fasta', checkIfExists: true)
                input[2] = true
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { with(workflow.out) {
                    assert snapshot(
                        versions.collect{ file(it) },
                        ch_reads_all[0][1].collect { file(it) },
                        ch_reads_fusioncatcher[0][1].collect { file(it) },
                        file(ch_fastp_html[0][1]).name,
                        file(ch_fastp_json[0][1]),
                        ch_fastqc_trimmed[0][1].collect { file(it).name }
                    ).match() }
                }
            )
        }
    }

    test("TRIM_WORKFLOW - Homo sapiens - FASTQs - fastp == false") {

        when {

            workflow {
                """
                // ch_reads
                input[0] = Channel.of(
                    [
                        [id: "test_fastq"],
                        [
                            file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/illumina/fastq/test_1.fastq.gz", checkIfExists: true),
                            file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/illumina/fastq/test_2.fastq.gz", checkIfExists: true)
                        ]
                    ])
                input[1] = Channel.empty()
                input[2] = false
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { with(workflow.out) {
                    assert snapshot(
                        versions.size() == 0,
                        ch_reads_all[0][1].size() == 2,
                        ch_reads_fusioncatcher[0][1].size() == 2,
                        ch_fastp_html.size() == 0,
                        ch_fastp_json.size() == 0,
                        ch_fastqc_trimmed.size() == 0
                    ).match() }
                }
            )
        }
    }

}
