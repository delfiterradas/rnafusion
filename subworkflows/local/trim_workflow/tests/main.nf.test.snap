{
    "TRIM_WORKFLOW - Homo sapiens - FASTQs - fastp == true": {
        "content": [
            [
                "versions.yml:md5,16187796d989b6260f572247e7dc0fc6",
                "versions.yml:md5,ea42abe9875f41f8362a55ee7533f102"
            ],
            [
                "test_fastq_1.fastp.fastq.gz:md5,0c436583301dea48755a5252a2675b64",
                "test_fastq_2.fastp.fastq.gz:md5,f7f38138255e63b33286b819b6177612"
            ],
            [
                "test_fastq_1.fastp.fastq.gz:md5,0c436583301dea48755a5252a2675b64",
                "test_fastq_2.fastp.fastq.gz:md5,f7f38138255e63b33286b819b6177612"
            ],
            "test_fastq.fastp.html",
            "test_fastq.fastp.json:md5,62066ad48c3d5981045cdd43e354cb2b",
            [
                "test_fastq_trimmed_1_fastqc.zip",
                "test_fastq_trimmed_2_fastqc.zip"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-05T16:21:52.926289296"
    },
    "TRIM_WORKFLOW - Homo sapiens - FASTQs - fastp == false": {
        "content": [
            true,
            true,
            true,
            true,
            true,
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-05T16:22:45.877168833"
    },
    "TRIM_WORKFLOW - Homo sapiens - FASTQs - fastp == true + fasp_adaptors": {
        "content": [
            [
                "versions.yml:md5,16187796d989b6260f572247e7dc0fc6",
                "versions.yml:md5,ea42abe9875f41f8362a55ee7533f102"
            ],
            [
                "test_fastq_1.fastp.fastq.gz:md5,adc67a7b4d0bf3520866d7599a4ba814",
                "test_fastq_2.fastp.fastq.gz:md5,9ee7d6c5230442970997477464255e67"
            ],
            [
                "test_fastq_1.fastp.fastq.gz:md5,adc67a7b4d0bf3520866d7599a4ba814",
                "test_fastq_2.fastp.fastq.gz:md5,9ee7d6c5230442970997477464255e67"
            ],
            "test_fastq.fastp.html",
            "test_fastq.fastp.json:md5,feb3483311bfa4ded60146f1cbc13fd5",
            [
                "test_fastq_trimmed_1_fastqc.zip",
                "test_fastq_trimmed_2_fastqc.zip"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-05T16:22:26.29488483"
    }
}